src/components/DeleteManifestConfirmationModal/__tests__/DeleteManifestConfirmationModal.test.tsx

```diff
@@ -0,0 +1,46 @@
+import React from 'react';
+import { render } from '@testing-library/react';
+import DeleteManifestConfirmationModal from '../DeleteManifestConfirmationModal';
+import useDeleteSatelliteManifest from '../../../hooks/useDeleteSatelliteManifest';
+
+jest.mock('../../../hooks/useDeleteSatelliteManifest');
+
+describe('DeleteConfirmationModal', () => {
+  const props = {
+    isOpen: true,
+    uuid: '00000000-0000-0000-0000-000000000000',
+    name: 'Manifest Name',
+    handleModalToggle: (): any => undefined,
+    onSuccess: (): any => undefined
+  };
+  let isLoading = false;
+
+  beforeEach(() => {
+    (useDeleteSatelliteManifest as jest.Mock).mockImplementation(() => ({
+      isLoading: isLoading,
+      isSuccess: false,
+      isError: false,
+      reset: (): any => undefined
+    }));
+  });
+
+  it('renders correctly', () => {
+    render(
+      <div>
+        <DeleteManifestConfirmationModal {...props} />
+      </div>
+    );
+    expect(document.body).toMatchSnapshot();
+  });
+
+  describe('when performing a delete', () => {
```

Comment:
One problem I'm having is that in rspec there is a `context` alias for `describe`, so in rspec a "when" statement like this would be in a context block and everything would be nicer to read.

As for what you're talking about, I've heard many people say to avoid deep nesting.  There is even a linting rule for that in our ruby linter.  This, however, is not deep nesting.  One describe for basically stating the subject and one would-be-context (describe) to set up conditions for that subject.  If these tests were more extensive, this tiny bit of encapsulation could be preventing tons of repetitive boiler plate and making it far easier to understand the test file.

I'll read the blog post though.

Reply:
kk cool yeah not a big deal. thx!